### PR TITLE
add smoke workflow

### PR DESCRIPTION
**Pull request by sarnold**

### CODE REVIEW

File: src/redis_ipc.c [hunk 1]
```diff
@@ -381,7 +379,7 @@ static json_object * redis_pop(const char *queue_path, unsigned int timeout)
 
     // parse popped entry back into json object
     entry = json_tokener_parse(json_text);
-    if (is_error(entry)) entry = NULL;
+    if (!entry) entry = NULL;
```
Comment (SJLC) [inline src/redis_ipc.c:384]: If json funcs nowadays just return NULL instead of an error that needs interpreted, just remove lines like this

**Comment by sarnold** [comment]
NOTE: this still has failing tests:

`command_result_test` fails on the first 3
`json_test` fails on focal only

* gentoo: dev-libs/json-c-0.15
* bionic: 0.12.1 flagged with **security**
* focal: 0.13.1 flagged with **security**  <= dumps core on json_test
* groovy: 0.15-1

**Comment by SJLC** [comment]
>     * focal: 0.13.1 flagged with **security**

How accurate is that security flag? I'm guessing it ignores CVE fix patches and just goes by the upstream version number, because the CVE reported against 0.13.1 has supposedly been addressed in focal
https://ubuntu.com/security/notices/USN-4360-4

**Comment by sarnold** [comment]
That's just the way they list their packages: https://packages.ubuntu.com/search?searchon=sourcenames&keywords=json-c

**Comment by SJLC** [comment]
Oh that's an indicator that you should get that package from security updates repo rather than original install media, is that what you meant above? I thought you meant those ones had known security bugs (CVE without corresponding patch).

**Comment by sarnold** [comment]
oh, I can actually see more in a more-old build, `json_object_object_get()` needs to migrate to `json_object_object_get_ex()`, as shown in bionic logs:

```
In file included from json_test.cpp:2:0:
../inc/json.hh: In member function ‘json json::get_field(const char*)’:
../inc/json.hh:56:72: warning: ‘json_object* json_object_object_get(json_object*, const char*)’ is deprecated [-Wdeprecated-declarations]
         json_object *field_obj = json_object_object_get(obj, field_name);
                                                                        ^
In file included from /usr/include/json-c/linkhash.h:16:0,
                 from /usr/include/json-c/json.h:22,
                 from ../inc/json.hh:4,
                 from json_test.cpp:2:
/usr/include/json-c/json_object.h:290:1: note: declared here
 THIS_FUNCTION_IS_DEPRECATED(extern struct json_object* json_object_object_get(struct json_object* obj,
 ^
```

**Pull request closed without merging**